synapse/util/caches/deferred_cache.py

```diff
@@ -282,11 +283,11 @@ def eb(_fail) -> None:
 
     def prefill(
         self, key: KT, value: VT, callback: Optional[Callable[[], None]] = None
-    ):
+    ) -> None:
         callbacks = [callback] if callback else []
         self.cache.set(key, value, callbacks=callbacks)
 
-    def invalidate(self, key):
+    def invalidate(self, key: KT) -> None:
```

Comment:
I'm not sure this actually accurate when using `TreeCache` and tuple keys, as I think we support giving a subset of the keys to delete multiple entries at once. I don't know how we type that though, so I'm tempted to just leave it for now.

Reply:
Agreed, let's ditch it